CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+# So You Want to Contribute to Papermill...
+We welcome all contributions to Papermill both large and small. We encourage you
+to join our community.
+
+## Our Community Values
+
+We are an open and friendly community. Everybody is welcome.
+
+We encourage friendly discussions and respect for all. There are no exceptions.
+
+All contributions are equally important. Documentation, answering questions, and
+fixing bugs are equally as valuable as adding new features.
+
+Check out <https://www.python.org/psf/codeofconduct/> for the Python code of conduct.
+
+## Setting up Your Development Environment
+Following these instructions should give you an efficient path to opening your first pull-request.
+
+### Cloning the Papermill Repository
+Fork the repository to your local Github account. Clone this repository to your local development machine.
+```
+git clone https://github.com/<your_account>/papermill
+cd papermill
+```
+
+### Install an Editable Version
+We prefer to use [conda](https://conda.io/docs/user-guide/tasks/manage-environments.html) to manage the development environment. 
+```
+conda create -n dev
+. activate env
+``` 
+or [virtualenv](https://packaging.python.org/guides/installing-using-pip-and-virtualenv/) if you prefer.
+```
+python3 -m virtualenv dev
+source dev/bin/activate 
+```
+
+```
+pip install -e .
```

Comment:
If we're going to encourage a dev install, then we should probably have people install their tests here rather than below. E.g.,

```
pip install -e .[dev]
```